### PR TITLE
Extract the route processing core from PippoFilter and implement a PippoServlet

### PR DESCRIPTION
**Pull request by gitblit**

### CODE REVIEW

File: pippo-core/src/main/java/ro/pippo/core/PippoConstants.java [hunk 1]
```diff
@@ -67,6 +73,45 @@
 
     public static final String UTF8 = StandardCharsets.UTF_8.toString();
 
+    public static final String PIPPO_LOGO = "\n"
+        + " ____  ____  ____  ____  _____\n"
+        + "(  _ \\(_  _)(  _ \\(  _ \\(  _  )\n"
+        + " ) __/ _)(_  ) __/ ) __/ )(_)(   http://pippo.ro\n"
+        + "(__)  (____)(__)  (__)  (_____)  {}\n";
+
+    public static String getPippoLogo() {
```
Comment (decebals) [inline pippo-core/src/main/java/ro/pippo/core/PippoConstants.java]: I propose you to move this method in a new class `PippoUtils` - for example `PippoUtils.getLogo()`, `PippoUtils.getVersion()` - and to keep in `PippoCostants` only constants.

File: pippo-core/src/main/java/ro/pippo/core/route/RouteDispatcher.java [hunk 2]
```diff
@@ -0,0 +1,233 @@
...
+
+    protected Router router;
+
+    protected ErrorHandler errorHandler;
+
+    public static <T extends RouteContext> T getRouteContext() {
+        return (T) ROUTE_CONTEXT_THREAD_LOCAL.get();
+    }
+
+    public RouteDispatcher() {
```
Comment (decebals) [inline pippo-core/src/main/java/ro/pippo/core/route/RouteDispatcher.java]: add Application as parameter in constructor and remove `setApplication()`?

File: pippo-core/src/main/java/ro/pippo/core/PippoServlet.java [hunk 3]
```diff
@@ -0,0 +1,134 @@
...
+        log.info("Pippo started ({})", runtimeMode);
+    }
+
+    @Override
+    public void service(ServletRequest req, ServletResponse resp) throws IOException, ServletException {
+
+        HttpServletRequest httpRequest = (HttpServletRequest) req;
+        HttpServletResponse httpResponse = (HttpServletResponse) resp;
+
+        Request request = new Request(httpRequest, routeDispatcher.getApplication());
```
Comment (decebals) [inline pippo-core/src/main/java/ro/pippo/core/PippoServlet.java]: why not `Request request = new Request(httpRequest, application);` ?

File: pippo-core/src/main/java/ro/pippo/core/PippoServlet.java [hunk 4]
```diff
@@ -0,0 +1,134 @@
...
+        Response response = new Response(httpResponse, routeDispatcher.getApplication());
```
Comment (decebals) [inline pippo-core/src/main/java/ro/pippo/core/PippoServlet.java]: same as above

**Comment by gitblit** [comment]
Changes pushed.

**Comment by decebals** [comment]
Great. Thanks

**Pull request merged**